### PR TITLE
[Mocha] Add tests

### PR DESCRIPTION
**Pull request by Its-Just-Nans**
add suppliers tests

### CODE REVIEW

File: test/supplier.test.js [hunk 1]
```diff
@@ -67,9 +110,27 @@ describe('Suppliers', () => {
...
+            expect(res.body).have.property('code');
+            expect(res.body.code).to.be.equal('Unauthorized');
+        });
+        it('Try delete a supplier - w/o the good ID', async () => {
+            await createSuppliers();
+            const res = await chai.request(app)
+                .delete('/api/v2/supplier/111111111111111111111111')
+                .set('authorization', credentials.token);
+            expect(res).to.have.status(200);
+            // TODO fix services suplliers to not send back 200 but use 'findOneAndDelete'
```
Comment (Its-Just-Nans) [inline test/supplier.test.js:133]: Is this a normal comportement ?
@devaquila 

(check the return of `deleteSupplier` in `services/suppliers.js:33`)

File: services/pictos.js [hunk 2]
```diff
@@ -51,8 +51,11 @@ const createPicto = async (picto) => {
 };
 
 const deletePicto = async (id) => {
-    const result = await Pictos.findOneAndRemove({_id: id});
-    const rule   = await ServiceRules.queryRule({filter: {owner_id: id}});
+    const result = await Pictos.findOneAndRemove({_id: id._id});
+    if (!result) {
```
Comment (Its-Just-Nans) [inline services/pictos.js:55]: ⚠️ change the service

File: services/pictos.js [hunk 3]
```diff
@@ -51,8 +51,11 @@ const createPicto = async (picto) => {
...
+        throw NSErrors.PictoNotFound;
+    }
```
Comment (Its-Just-Nans) [inline services/pictos.js:57]: ⚠️ change the service

File: services/slider.js [hunk 4]
```diff
@@ -48,11 +48,13 @@ const createSlider = async (req) => {
 const deleteSlider = async (req) => {
     if (!mongoose.Types.ObjectId.isValid(req.params.id)) throw NSErrors.UnprocessableEntity;
     const doc = await Slider.findOneAndRemove({_id: req.params.id});
+    if (!doc) {
+        throw NSErrors.NotFound;
+    }
```
Comment (Its-Just-Nans) [inline services/slider.js:53]: ⚠️ change the service

-> no custom error ? (like `NSErrors.NotFound;`)

File: utils/errors/NSErrors.js [hunk 5]
```diff
@@ -118,6 +118,7 @@ class NSErrors {
     static get ResetPasswordMailContentAdminNotExists() { return this.makeErrorAlias(NSErrors.NotFound, 'ResetPasswordMailContentAdminNotExists'); }
     static get SetAttributeNotFound() { return this.makeErrorAlias(NSErrors.NotFound, 'SetAttributeNotFound'); }
     static get ShipmentUpdateError() { return this.makeErrorAlias(NSErrors.NotFound, 'ShipmentUpdateError'); }
+    static get ShipmentNotFound() { return this.makeErrorAlias(NSErrors.NotFound, 'ShipmentNotFound'); }
```
Comment (Its-Just-Nans) [inline utils/errors/NSErrors.js:121]: Create a new Error

File: test/bills.test.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        });
+    });
+    describe('POST /api/v2/bills/generatePDF', () => {
+        it('Generate PDF of bills', async () => {
+            const bill = await createBills();
+            const res  = await chai.request(app)
+                .post('/api/v2/bills/generatePDF')
+                .set('authorization', credentials.token)
+                .send({PostBody: {structure: {__v: 0}, filter: {_id: bill._id}}});
+            expect(res).to.have.status(500);
```
Comment (Its-Just-Nans) [inline test/bills.test.js:57]: generatePDF don't work

File: services/gallery.js [hunk 7]
```diff
@@ -117,11 +117,13 @@ const setItemGallery = async (_id, datas) => {
 const deleteGallery = async (_id) => {
     if (!mongoose.Types.ObjectId.isValid(_id)) throw NSErrors.InvalidObjectIdError;
     const doc = await Gallery.findOneAndRemove({_id});
+    if (!doc) {
+        throw NSErrors.GalleryNotFound;
+    }
```
Comment (Its-Just-Nans) [inline services/gallery.js:122]: Change the service

File: routes/orders.js [hunk 8]
```diff
@@ -62,6 +62,7 @@ async function getOrders(req, res, next) {
 async function setOrder(req, res, next) {
     // On update la commande
     try {
+        console.log(req.body.order);
```
Comment (haegemonia76) [inline routes/orders.js:65]: remove `console.log`

File: services/slider.js (hunk 4 above)
Comment (haegemonia76) [reply services/slider.js:53]: You should create one

File: test/bills.test.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+        it('Create bill and get it with the code', async () => {
+            const bill = await createBills();
+            const res  = await chai.request(app)
+                .post('/api/v2/bills')
+                .set('authorization', credentials.token)
+                .send({PostBody: {filter: {code: bill.code}, limit: 99}});
+            expect(res).to.have.status(200);
+            expect(res.body.datas[0].name).be.equals(bill.name);
+        });
+        it('Create bill and get it with the id - w/o authentication', async () => {
```
Comment (haegemonia76) [inline test/bills.test.js:30]: The description does not match your test
Should be something like this : `it('Should not retrieve bill if no authorization token send', ...`

File: test/bills.test.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+            expect(res.body).have.property('code');
+            expect(res.body.code).to.be.equal('AccessUnauthorized');
+        });
+        it('Create bill and get it with the id - w/o the good id', async () => {
+            await createBills();
+            const res = await chai.request(app)
+                .post('/api/v2/bills')
+                .set('authorization', credentials.token)
+                .send({PostBody: {filter: {_id: '111111111111111111111111'}, limit: 99}});
+            expect(res).to.have.status(200);
```
Comment (haegemonia76) [inline test/bills.test.js:45]: Weird, I expect to have a 404 not a 200 when sending wrong information.
If the api send weird thing don't follow blindly what the api returns to you

File: test/bills.test.js [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+        it('Create bill and get it with the id - w/o authentication', async () => {
+            const bill = await createBills();
+            const res  = await chai.request(app)
+                .post('/api/v2/bills')
+                .send({PostBody: {filter: {_id: bill._id}, limit: 99}});
+            expect(res).to.have.status(403);
+            expect(res.body).have.property('code');
+            expect(res.body.code).to.be.equal('AccessUnauthorized');
+        });
+        it('Create bill and get it with the id - w/o the good id', async () => {
```
Comment (haegemonia76) [inline test/bills.test.js:39]: `it('Should create a bill and don't get it when sending a wrong id', ...`

File: test/bills.test.js [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+    describe('POST /api/v2/bills/generatePDF', () => {
+        it('Generate PDF of bills', async () => {
+            const bill = await createBills();
+            const res  = await chai.request(app)
+                .post('/api/v2/bills/generatePDF')
+                .set('authorization', credentials.token)
+                .send({PostBody: {structure: {__v: 0}, filter: {_id: bill._id}}});
+            expect(res).to.have.status(500);
+        });
+        it('Create bill and get it with the id - w/o authentication', async () => {
```
Comment (haegemonia76) [inline test/bills.test.js:59]: not good description

File: test/categories.test.js [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+        });
+        it('Create category and get it with the id - w/o authentication', async () => {
+            const category = await createCategory();
+            const res      = await chai.request(app)
+                .post('/api/v2/categories')
+                .send({PostBody: {filter: {_id: category._id}, limit: 99}});
+            expect(res).to.have.status(200);
+            expect(res.body.datas[0].code).be.equals(category.code);
+        });
+        it('Create category and get it with the id - w/o the good id', async () => {
```
Comment (haegemonia76) [inline test/categories.test.js:39]: not good description

File: test/categories.test.js [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+            await createCategory();
+            const res = await chai.request(app)
+                .post('/api/v2/categories')
+                .set('authorization', credentials.token)
+                .send({PostBody: {filter: {_id: '111111111111111111111111'}, limit: 99}});
+            expect(res).to.have.status(200);
```
Comment (haegemonia76) [inline test/categories.test.js:45]: weird behavior

File: test/categories.test.js [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+        it('Create category and get it with the id', async () => {
+            const category = await createCategory();
+            const res      = await chai.request(app)
+                .post('/api/v2/categories')
+                .set('authorization', credentials.token)
+                .send({PostBody: {filter: {_id: category._id}, limit: 99}});
+            expect(res).to.have.status(200);
+            expect(res.body.datas[0].code).be.equals(category.code);
+        });
+        it('Create category and get it with the id - w/o authentication', async () => {
```
Comment (haegemonia76) [inline test/categories.test.js:31]: not good description

**Review by haegemonia76 — CHANGES_REQUESTED** [review]
The description of your tests are not good, they needs to explicitly say what you are testing.

Some test are weird, they return 200 even when they clearly should return a 404 and obviously not pass

File: test/bills.test.js (hunk 10 above)
Comment (devaquila) [reply test/bills.test.js:45]: Yes. Big change for that, don't change it please

File: test/categories.test.js (hunk 14 above)
Comment (devaquila) [reply test/categories.test.js:45]: Yes. Big change for that, don't change it please

**Review by haegemonia76 — CHANGES_REQUESTED** [review]
Just add 'should' at the start does not fix bad description of your tests, write correct english sentence

**Pull request merged**